### PR TITLE
Clean and refactor

### PR DESCRIPTION
**Pull request by sandlerben**
See commit messages for details

### CODE REVIEW

**Comment by sandlerben** [comment]
r @yoninachmany @rachelhongca

File: transcription/transcription.go [hunk 1]
```diff
@@ -11,13 +11,15 @@ import (
 
 // SendEmail connects to an email server at host:port, switches to TLS,
 // authenticates on TLS connections using the username and password, and sends
-// an email from address from, to address to, with subject line subject with message body.
+// an email from address from, to address to, with subject line subject with
+// message body.
 func SendEmail(username string, password string, host string, port int, to []string, subject string, body string) error {
```
Comment (yoninachmany) [inline transcription/transcription.go:18]: Would it be "cleaner" if we made this fit into character limit by breaking up into two lines?

**Comment by yoninachmany** [comment]
Can you just explain motivation for changing to concurrent map struct? Thanks.

File: transcription/transcription.go (hunk 1 above)
Comment (sandlerben) [inline transcription/transcription.go:18]: So Effective Go says

> Go has no line length limit. Don't worry about overflowing a punched card. If a line feels too long, wrap it and indent with an extra tab.

I read or was told once to try to keep comments below 80 characters though. I think it's our call if we want to break the line up.

Comment (yoninachmany) [inline transcription/transcription.go:18]: I think it's ok.

**Comment by sandlerben** [comment]
@yoninachmany I didn't like that the executer itself contained the `Mutex`. It's more idiomatic to make a custom map which contains a `Mutex`.

For example, why should the executer itself be "locked" when we just want to read from it's map. It's better to just lock the map itself.

File: tasks/tasks.go [hunk 2]
```diff
@@ -36,6 +40,21 @@ const (
 // DefaultTaskExecuter is an instance of a NewTaskExecuter.
 var DefaultTaskExecuter = NewTaskExecuter()
 
+// Map k to v in the map
```
Comment (sandlerben) [inline tasks/tasks.go]: Oops I'll fix these function comments

Comment (yoninachmany) [inline tasks/tasks.go]: Do we need these function comments?

Comment (sandlerben) [inline tasks/tasks.go]: Technically no because these functions are unexported but I still like to have comments.

**Comment by sandlerben** [comment]
@yoninachmany can you look at the changes I've made since you reviewed? 

@rachelhongca can you give this a look?

File: tasks/tasks.go [hunk 3]
```diff
@@ -40,14 +40,14 @@ const (
 // DefaultTaskExecuter is an instance of a NewTaskExecuter.
 var DefaultTaskExecuter = NewTaskExecuter()
 
-// Map k to v in the map
+// put maps k to v in the map
```
Comment (yoninachmany) [inline tasks/tasks.go]: nitpick but would change to "put(k,v)"

File: tasks/tasks.go [hunk 4]
```diff
@@ -40,14 +40,14 @@ const (
...
 func (c *concurrentStatusMap) put(k string, v Status) {
 	c.Lock()
 	c.m[k] = v
 	c.Unlock()
 }
 
-// Get the value of k in the map
+// get returns the value of k in the map
```
Comment (yoninachmany) [inline tasks/tasks.go]: get(k)

**Comment by yoninachmany** [comment]
Can you explain the change from w.Write to io.WriteString? Other than that, lgtm.

**Comment by rachelhongca** [comment]
lgtm!

**Pull request merged**